boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -43,24 +44,16 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
             has_charger_faulted = App_Charger_HasFaulted(charger);
         }
 
-        uint8_t    segment = 0U;
-        uint8_t    cell    = 0U;
-        const bool has_reached_max_v =
-            App_Accumulator_GetMaxVoltage(accumulator, &segment, &cell) > MAX_CELL_VOLTAGE_THRESHOLD;
+        const bool charging_completed = App_TractiveSystem_GetCurrent(ts) <= CURRENT_AT_MAX_CHARGE;
 
-        App_CanTx_BMS_Faults_ChargerDisconnectedInChargeState_Set(is_charger_disconnected);
         App_CanTx_BMS_Faults_ChargerFault_Set(has_charger_faulted);
-        App_CanTx_BMS_Charger_IsChargingComplete_Set(has_reached_max_v);
```

Comment:
IsChargingComplete and ChargerDisconnectedInChargeState are important signals - you should change this back so they are actually being set

Reply:
Also the faults here don't actually put the board into fault state